File: src/main/java/com/smartcampus/booking/service/BookingService.java
package com.smartcampus.booking.service;

import com.smartcampus.booking.dto.BookingRequestDTO;
import com.smartcampus.booking.dto.BookingResponseDTO;
import java.util.List;

public interface BookingService {
    BookingResponseDTO createBooking(BookingRequestDTO dto);
    BookingResponseDTO approveBooking(String id);
    BookingResponseDTO rejectBooking(String id, String reason);
    BookingResponseDTO cancelBooking(String id);
    BookingResponseDTO getBookingById(String id);
    List<BookingResponseDTO> getBookingsByUser(String userId);
    List<BookingResponseDTO> getAllBookings();
}

File: src/services/bookingService.js
import axios from 'axios';

const api = axios.create({
  baseURL: '/api',
  headers: { 'Content-Type': 'application/json' },
});

const bookingService = {
  getAllBookings: () => api.get('/bookings'),
  getBookingById: (id) => api.get(`/bookings/${id}`),
  getBookingsByUser: (userId) => api.get(`/bookings/user/${userId}`),
  createBooking: (data) => api.post('/bookings', data),
  approveBooking: (id) => api.patch(`/bookings/${id}/approve`),
  rejectBooking: (id, reason) => api.patch(`/bookings/${id}/reject`, { reason }),
  cancelBooking: (id) => api.patch(`/bookings/${id}/cancel`),
};

export default bookingService;